client/state/selectors/is-connected-secondary-network-site.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import createSelector from 'lib/create-selector';
+import { isMainSiteOf } from 'state/selectors';
+import { isJetpackSiteSecondaryNetworkSite } from 'state/sites/selectors';
+
+/**
+ * Returns true if site with id equal to siteId is a connected secondary network site and false otherwise
+ * Returns null if with the information available it's not possible to determine if site is a connected secondary site or not
+ * In order for a site to be a connected secondary site its main site also needs to be connected
+ * With selector isJetpackSiteSecondaryNetworkSite the main site does not need to be connected
+ *
+ * @param  {Object}    state     Global state tree
+ * @param  {Number}    siteId    The ID of the site we're querying
+ * @return {?Boolean}            Whether site with id equal to siteId is a connected secondary network site
+ */
+export default createSelector(
+	( state, siteId ) => {
+		const isSecondaryNetworkSite = isJetpackSiteSecondaryNetworkSite( state, siteId );
+		if ( isSecondaryNetworkSite === null ) {
+			return null;
+		}
+		if ( ! isSecondaryNetworkSite ) {
+			return false;
+		}
+		const siteIds = Object.keys( get( state, 'sites.items', {} ) );
+		return siteIds.reduce( ( previousResult, mainSiteId ) => {
+			if ( previousResult ) {
+				return previousResult;
```

Comment:
Hi @tyxla, you are right one of my tests was not properly testing the case as it should and that was improved 👍 
The equivalent version with lodash some to my reduce code would be:
```
		if ( some( siteIds, ( mainSiteId ) => isMainSiteOf( state, mainSiteId, siteId ) ) ) {
			return true;
		}
		if ( some( siteIds, ( mainSiteId ) => isMainSiteOf( state, mainSiteId, siteId ) === null ) ) {
			return null;
		}
		return false;
```
The reason why this selector returns null its because its a common practice that selectors return null if its not possible to determine the output. In this case, it's not possible to determine if non of the isMainSiteOf returned true and at least one returned null. But until now there is no practical use of null in this selector and if you prefer I can make it return just true or false.